### PR TITLE
#169540093- Fetch data

### PR DESCRIPTION
**Pull request by mlodydev**
Double 'return' is to be repaired.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
   },
   "dependencies": {
     "react": "16.9.0",
-    "react-native": "0.61.3"
+    "react-native": "0.61.3",
+    "react-native-webview": "^7.4.3"
```
Comment (Lukkub) [inline package.json]: Remove unused dependencies

File: App.js [hunk 2]
```diff
@@ -1,58 +1,78 @@
...
+  },
+});
 
 class App extends Component{
+  constructor(props){
+    super(props);
+
+    this.state = {
+      data: null,
+      isLoading: true,
```
Comment (Lukkub) [inline App.js]: It will be safe pattern to set isLoading on `true` when fetchApiData method is called.

File: App.js [hunk 3]
```diff
@@ -1,58 +1,78 @@
...
   FlatList,
+  Text,
+  ActivityIndicator,
+  StyleSheet
 } from 'react-native';  
 
 import ListItem from './modules/ListItem';
 
-const dataUrl = 'https://picsum.photos/v2/list';
+const DATAURL = 'https://picsum.photos/v2/list';
```
Comment (Lukkub) [inline App.js]: `DATAURL` ===> `DATA_URL`

File: App.js [hunk 4]
```diff
@@ -1,58 +1,78 @@
...
+    .catch((error) => console.warn(error));
+  }
+
+  componentDidMount(){
+    this.fetchApiData();
+  }
+
   render(){
+    let currentView;
+    if(this.state.isLoading){
```
Comment (Lukkub) [inline App.js]: It will be better to use operator `? :`to set this condition. Code should be simpler and clean.

File: App.js [hunk 5]
```diff
@@ -1,58 +1,78 @@
...
+      currentView = <View style={styles.loadingIndicator}>
+      <ActivityIndicator size='large' color='#058cd9' show={this.state.isLoading}/>
+    </View>
+    }
+    else{
+    currentView = <View>
```
Comment (Lukkub) [inline App.js]: Do you need this `View` wrapper here ?

File: App.js [hunk 6]
```diff
@@ -1,58 +1,78 @@
...
 
 import ListItem from './modules/ListItem';
 
-const dataUrl = 'https://picsum.photos/v2/list';
+const DATAURL = 'https://picsum.photos/v2/list';
+
+const styles = StyleSheet.create({
+  loadingIndicator: {
+    flex: 1,
+    display: 'flex',
```
Comment (Lukkub) [inline App.js]: 'flex' is the default for display so Yo don't need to set up it.

File: App.js [hunk 7]
```diff
@@ -1,58 +1,78 @@
...
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  container:{
+    flex: 1,
+    display: 'flex',
```
Comment (Lukkub) [inline App.js]: 'flex' is the default for display so Yo don't need to set up it.

File: modules/ListItem.js [hunk 8]
```diff
@@ -62,24 +63,40 @@ const styles = StyleSheet.create({
...
-                    <View style={styles.idView}>
-                        <Text style={styles.id}>{props.id}</Text>
+class ListItem extends Component{
+    constructor(props){
+        super(props);
+        this.onPressHandler = this.onPressHandler.bind(this);
+    }
+
+    onPressHandler(){
+        try{
```
Comment (Lukkub) [inline modules/ListItem.js]: Here you use `try catch` but on the parent component u use `then catch`. Please be consistent in the code writing.

**Review by Lukkub — CHANGES_REQUESTED** [review]
Few things to clean here. Let me know if you have any problems.

File: App.js (hunk 4 above)
Comment (mlodydev) [reply App.js]: As it is not one line of code i thought that it will be more clear with if.

Comment (Lukkub) [reply App.js]: you can do sth like that that is much clearer.
```
const currentView = this.state.isLoading 
? COMPONENT_X
: COMPONENT_Y
```

so you do not overwrite `currentView` variable.

**Comment by mlodydev** [comment]
Changes applied. Review and accept needed. @Lukkub

File: App.js (hunk 2 above)
Comment (Lukkub) [reply App.js]: Please change initial value to false

File: App.js [hunk 9]
```diff
@@ -1,58 +1,72 @@
...
-            id: '4'},
-            {imageUrl: "https://statics.sportskeeda.com/editor/2018/03/a4a7b-1520474015-800.jpg", 
-            name: 'Michaelsadkoasdkashdsauo Jordan12345y78781212313812378', 
-            pageUrl: 'whatever.com/whatever.com/whatever.com/whatever.com/whatever.com/whatever.com/whatever.com/whatever.com', 
-            id: '5'},
-          ]}
-          renderItem={({item}) => <ListItem id={item.id} imageUrl={item.imageUrl} name={item.name} pageUrl={item.pageUrl}/>}
-        />
+      <View style={styles.container}>  
+        {loadingOrList}
```
Comment (Lukkub) [inline App.js]: Just name `body` is enough.

**Review by Lukkub — CHANGES_REQUESTED** [review]

**Comment by mlodydev** [comment]
I hope it's good. @Lukkub

**Review by Lukkub — APPROVED** [review]

**Pull request merged**